### DISCUSSION TITLE
Clearing cache directly via a url?

### DISCUSSION THREAD

**Discussion by ogcreid** [wp-media/wp-rocket#6161] [Q&A]
According to Chat GPT, this code, inserted into `functions.php` for a Wordpress site, will clear the cache if I execute 

https://mysite.com/?secret_purge=YOUR_SECRET_CODE

`function custom_purge_wprocket() {
    if ( isset( $_GET['secret_purge'] ) && 'YOUR_SECRET_CODE' == $_GET['secret_purge'] ) {
        if ( function_exists( 'rocket_clean_domain' ) ) {
            rocket_clean_domain();
            echo 'WP Rocket cache cleared!';
        } else {
            echo 'WP Rocket is not activated.';
        }
        exit;
    }
}
add_action( 'init', 'custom_purge_wprocket' );`

Will that work?

**Comment by andrewinsidelazarev** [comment]
add_action('rest_api_init', function () {
register_rest_route('wp-rocket-cache/v1', '/clear/', array(
'methods' => 'POST',
'callback' => 'clear_wp_rocket_cache_with_token',
));
});

function clear_wp_rocket_cache_with_token(WP_REST_Request $request) {
// Получаем токен из заголовка или параметра запроса
$provided_token = $request->get_header('X_CACHE_TOKEN') ?: $request->get_param('token');
$valid_token = '[your_token]'; // Замените на свой

if (empty($provided_token) || $provided_token !== $valid_token) {
    return new WP_Error('invalid_token', 'Invalid or missing token', array('status' => 403));
}

if (!function_exists('rocket_clean_domain')) {
    return new WP_Error('wp_rocket_missing', 'WP Rocket is not active', array('status' => 500));
}

rocket_clean_domain(); // Очистка кэша
return rest_ensure_response(['success' => true, 'message' => 'Cache cleared']);
}

**Reply by andrewinsidelazarev** [reply]
It's works for me
